### DISCUSSION TITLE
How to enable AVX2 or AVX512 on CPU with both vector intrinsics

### DISCUSSION THREAD

**Discussion by jiangkai1109** [milvus-io/milvus#25626] [Q&A and General discussion] [answered]
I am benchmarking AVX2 vs AVX512 on Milvus. CPU supports both vector intrinsics. 
May I know how to enable AVX2 or AVX512 dedicatedly during milvus standalone(docker and dev mode) runtime? Or I need to do this during build time, such as add avx2 parameters to cmake during build period?

Below are steps how to run milvus standalone in docker and dev mode following guide in milvus web.

Start milvus standalone in docker or dev mode as below:
1) milvus standalone 
Or running Milvus standalone
$ cd deployments/docker/dev
$ docker-compose up -d
$ cd ../../../
$ ./scripts/start_standalone.sh

2) milvus docker
wget https://github.com/milvus-io/milvus/releases/download/v2.2.11/milvus-standalone-docker-compose.yml -O docker-compose.yml
sudo docker-compose up -d

**Comment by cydrain** [comment, marked as answer]
Hi @jiangkai1109 ,
you can set SIMD via configs/milvus.yaml
```
  # Default value: auto
  # Valid values: [auto, avx512, avx2, avx, sse4_2]
  # This configuration is only used by querynode and indexnode, it selects CPU instruction set for Searching and Index-building.
  simdType: auto
```

and when Milvus launch, you can find the log like following:
```
I20230714 12:32:38.603224 3191433 knowhere_config.cc:62] [KNOWHERE][SetSimdType][milvus] FAISS expect simdType::AUTO
I20230714 12:32:38.603237 3191433 knowhere_config.cc:87] [KNOWHERE][SetSimdType][milvus] FAISS hook AVX2
```

"FAISS hook AVX2" means Milvus selects AVX2

**Reply by jiangkai1109** [reply]
It works for me. After set simd_type from auto to avx2, per the log info, FAISS switch from avx512 to avx2. Thanks for help!

vus/milvus_code/milvus/configs/easylogging.yaml
2023-07-18 14:15:48,731 | DEBUG | default | [SEGCORE][SegcoreSetSimdType][milvus] set config simd_type: avx2
2023-07-18 14:15:48,731 | INFO | default | [KNOWHERE][SetSimdType][milvus] FAISS expect simdType::AVX2
2023-07-18 14:15:48,731 | INFO | default | [KNOWHERE][SetSimdType][milvus] FAISS hook AVX2
